autoendpoint/src/headers/vapid.rs

```diff
@@ -149,26 +149,23 @@ impl VapidHeader {
     /// WARNING: THIS FUNCTION DOES NOT VALIDATE THE VAPID HEADER AND SHOULD
     /// ONLY BE USED FOR LOGGING AND METRIC REPORTING FUNCTIONS.
     /// Proper validation should be done by [crate::extractors::subscription::validate_vapid_jwt()]
-    pub fn insecure_sub(&self) -> Result<String, VapidError> {
+    pub fn insecure_sub(&self) -> Result<Option<String>, VapidError> {
         // This parses the VAPID header string
         let data = VapidClaims::try_from(self.clone()).inspect_err(|e| {
             warn!("🔐 Vapid: {:?} {:?}", e, &self.token);
         })?;
 
-        if let Some(sub) = data.sub {
-            if !sub.starts_with("mailto:") && !sub.starts_with("https://") {
-                info!("🔐 Vapid: Bad Format {:?}", sub);
-                return Err(VapidError::SubBadFormat);
-            }
-            if sub.is_empty() {
-                info!("🔐 Empty Vapid sub");
-                return Err(VapidError::SubEmpty);
-            }
-            info!("🔐 Vapid: sub: {:?}", sub);
-            return Ok(sub.to_owned());
+        let Some(sub) = data.sub else { return Ok(None) };
+        if !sub.starts_with("mailto:") && !sub.starts_with("https://") {
+            info!("🔐 Vapid: Bad Format {sub:?}");
+            return Err(VapidError::SubBadFormat);
+        };
+        if sub.is_empty() {
+            info!("🔐 Empty Vapid sub");
```

Comment:
@jrconlin : This is unreachable code, sub can't start with "mailto:" or "https://" and be empty. The check for empty sub must be moved before the other one.

You can use tests from #830 to correctly check that

Reply:
Ah! Good catch. I'll submit a follow up.